tests/test_transforms/test_math/test_binary_operator_transform.py

```diff
@@ -0,0 +1,339 @@
+import operator
+from copy import deepcopy
+
+import numpy as np
+import numpy.testing
+import pytest
+
+from etna.datasets import TSDataset
+from etna.datasets import generate_ar_df
+from etna.transforms.math import binary_operator
+
+ops = {
+    "+": operator.add,
+    "-": operator.sub,
+    "*": operator.mul,
+    "/": operator.truediv,
+    "/=": operator.floordiv,
+    "%": operator.mod,
+    "**": operator.pow,
+    "==": operator.eq,
+    ">=": operator.ge,
+    "<=": operator.le,
+    ">": operator.gt,
+    "<": operator.lt,
+}
+
+
+@pytest.fixture
+def ts_one_segment(random_seed) -> TSDataset:
+    """Generate dataset with non-positive target."""
+    df = generate_ar_df(start_time="2020-01-01", periods=100, freq="D", n_segments=1)
+    df["feature"] = np.random.uniform(10, 0, size=100)
+    df["target"] = np.random.uniform(10, 0, size=100)
+    df = TSDataset.to_dataset(df)
+    ts = TSDataset(df, freq="D")
+    return ts
+
+
+@pytest.fixture
+def ts_two_segments(random_seed) -> TSDataset:
+    """Generate dataset with non-positive target."""
+    df = generate_ar_df(start_time="2020-01-01", periods=100, freq="D", n_segments=2)
+    df["feature"] = np.random.uniform(10, 0, size=200)
+    df["target"] = np.random.uniform(10, 0, size=200)
+    df = TSDataset.to_dataset(df)
+    ts = TSDataset(df, freq="D")
+    return ts
+
+
+@pytest.mark.parametrize(
+    "operand, left_column, right_column, out_column",
+    [
+        ("+", "feature", "target", "target"),
+        ("-", "feature", "target", "target"),
+        ("*", "feature", "target", "target"),
+        ("/", "feature", "target", "target"),
+        ("/=", "feature", "target", "target"),
+        ("%", "feature", "target", "target"),
+        ("**", "feature", "target", "target"),
+        ("==", "feature", "target", "target"),
+        (">=", "feature", "target", "target"),
+        ("<=", "feature", "target", "target"),
+        (">", "feature", "target", "target"),
+        ("<", "feature", "target", "target"),
+        ("+", "target", "feature", "target"),
+        ("-", "target", "feature", "target"),
+        ("*", "target", "feature", "target"),
+        ("/", "target", "feature", "target"),
+        ("/=", "target", "feature", "target"),
+        ("%", "target", "feature", "target"),
+        ("**", "target", "feature", "target"),
+        ("==", "target", "feature", "target"),
+        (">=", "target", "feature", "target"),
+        ("<=", "target", "feature", "target"),
+        (">", "target", "feature", "target"),
+        ("<", "target", "feature", "target"),
+        ("+", "feature", "target", "feature"),
```

Comment:
I don't understand why should we need `feature, target, feature`, because it checks almost the same thing as `target, feature, target`.

Reply:
in the initial version it was there, Yasha asked to add

Reply:
Yes, it's by bad.
For some reason I thought this needed to be added. It seems that I was wrong, let's delete it.